source/common/http/access_log/access_log_formatter.cc

```diff
@@ -104,14 +106,18 @@ static const std::string Http10String = "HTTP/1.0";
 static const std::string Http11String = "HTTP/1.1";
 static const std::string Http2String = "HTTP/2";
 
-const std::string& AccessLogFormatUtils::protocolToString(Protocol protocol) {
-  switch (protocol) {
-  case Protocol::Http10:
-    return Http10String;
-  case Protocol::Http11:
-    return Http11String;
-  case Protocol::Http2:
-    return Http2String;
+const std::string& AccessLogFormatUtils::protocolToString(Optional<Protocol> protocol) {
+  if (protocol.valid()) {
```

Comment:
is it possible to make this a bit more generic like protocol is TCP/mongo/redis/etc?

Reply:
This was tricky because there's one filter that's using the request_info to figure out whether it is http1 or http2.  I'd like to make this better, but I didn't want to have the scope of this increase too much.  I mostly wanted to get the core of tcp access logging in, and then we can enable logging more and more as we go.